### PR TITLE
Add implementation of package linting rule and tests

### PR DESCRIPTION
**Pull request by carlosliarte**

### CODE REVIEW

**Comment by wscheep** [comment]
Solves #29

File: core/src/main/java/io/anemos/metastore/core/proto/validate/ProtoLint.java [hunk 1]
```diff
@@ -166,4 +167,20 @@ private boolean isPascalCase(String fieldName) {
         return true;
     }
 
+    public void lintOnPackage(Descriptors.Descriptor ref) {
+        Descriptors.FileDescriptor fileDescriptor = proto.getFileDescriptorByFileName(ref.getFile().getName());
+        String protoPackageName = fileDescriptor.getPackage();
+        String fileName = ref.getFile().getFullName();
+        fileName = fileName.substring(0, fileName.lastIndexOf("/")).replace("/", ".");
```
Comment (alexvanboxel) [inline core/src/main/java/io/anemos/metastore/core/proto/validate/ProtoLint.java]: The result of the substring is a directory name. Use a new variable dirName to indicate what it is.

File: core/src/main/java/io/anemos/metastore/core/proto/validate/ProtoLint.java [hunk 2]
```diff
@@ -166,4 +167,20 @@ private boolean isPascalCase(String fieldName) {
...
+        if(!fileName.equals(protoPackageName)){
```
Comment (alexvanboxel) [inline core/src/main/java/io/anemos/metastore/core/proto/validate/ProtoLint.java]: Use your IntelliJ code formatter to make sure your code is correctly formatted. Add space before opening bracket.

File: proto/src/main/proto/anemos/metastore/v1alpha1/rules.proto [hunk 3]
```diff
@@ -27,4 +27,7 @@ enum LintRule {
     LINT_METHOD_RTYPE_END_WITH_RESPONSE = 50003;
 
     LINT_GENERAL_CUSTOM = 60000;
+
+    LINT_PACKAGE_CUSTOM = 70000;
+    LINT_PACKAGE_NAME_SHOULD_BE_VALID = 70001;
```
Comment (alexvanboxel) [inline proto/src/main/proto/anemos/metastore/v1alpha1/rules.proto]: change to LINT_PACKAGE_NO_DIR_ALIGNMENT

**Pull request merged**